include/picolibrary/microchip/megaavr0/multiplexed_signals/atmega4809.h

```diff
@@ -31,7 +39,480 @@
+ *
+ * \param[in] usart The USART peripheral whose XCK pin number is to be looked up.
+ *
+ * \return The USART peripheral's XCK pin number.
+ */
+inline auto xck_number( Peripheral::USART const & usart ) noexcept -> std::uint_fast8_t
+{
+    // #lizard forgives the length
+
+    switch ( reinterpret_cast<std::uintptr_t>( &usart ) ) {
+        case Peripheral::ATmega4809::USART0::ADDRESS:
+            switch ( Peripheral::ATmega4809::PORTMUX0::instance().usart0_route() ) {
+                case Peripheral::PORTMUX::USART_Route::DEFAULT: return 2;
+                case Peripheral::PORTMUX::USART_Route::ALTERNATE: return 6;
+                default: break;
+            } // switch
+            break;
+        case Peripheral::ATmega4809::USART1::ADDRESS:
+            switch ( Peripheral::ATmega4809::PORTMUX0::instance().usart1_route() ) {
+                case Peripheral::PORTMUX::USART_Route::DEFAULT: return 2;
+                case Peripheral::PORTMUX::USART_Route::ALTERNATE: return 6;
+                default: break;
+            } // switch
+            break;
+        case Peripheral::ATmega4809::USART2::ADDRESS:
+            switch ( Peripheral::ATmega4809::PORTMUX0::instance().usart2_route() ) {
+                case Peripheral::PORTMUX::USART_Route::DEFAULT: return 2;
+                case Peripheral::PORTMUX::USART_Route::ALTERNATE: return 6;
+                default: break;
+            } // switch
+            break;
+        case Peripheral::ATmega4809::USART3::ADDRESS:
+            switch ( Peripheral::ATmega4809::PORTMUX0::instance().usart3_route() ) {
+                case Peripheral::PORTMUX::USART_Route::DEFAULT: return 2;
+                case Peripheral::PORTMUX::USART_Route::ALTERNATE: return 2;
+                default: break;
+            } // switch
+            break;
+    } // switch
+
+    return 0;
+}
+
+/**
+ * \brief Lookup a USART peripheral's XCK pin mask.
+ *
+ * \attention This function should never be called directly. Instead, set the `-mmcu`
+ *            compiler flag to `atmega4809` and call
+ *            picolibrary::Microchip::megaAVR0::Multiplexed_Signals::xck_mask().
+ *
+ * \param[in] usart The USART peripheral whose XCK pin mask is to be looked up.
+ *
+ * \return The USART peripheral's XCK pin mask.
+ */
+inline auto xck_mask( Peripheral::USART const & usart ) noexcept -> std::uint8_t
+{
+    return 1 << xck_number( usart );
+}
+
+/**
+ * \brief Lookup a USART peripheral's XDIR pin port.
+ *
+ * \attention This function should never be called directly. Instead, set the `-mmcu`
+ *            compiler flag to `atmega4809` and call
+ *            picolibrary::Microchip::megaAVR0::Multiplexed_Signals::xdir_port().
+ *
+ * \param[in] usart The USART peripheral whose XDIR pin port is to be looked up.
+ *
+ * \return The USART peripheral's XDIR pin port.
+ */
+inline auto xdir_port( Peripheral::USART const & usart ) noexcept -> Peripheral::PORT &
```

Comment:
Return type can be deduced.
```suggestion
inline auto & xdir_port( Peripheral::USART const & usart ) noexcept
```